tests/framework/extensions/rbac/verify.go

```diff
@@ -0,0 +1,252 @@
+	actualClusterID := clusterStatus.ClusterName
+	assert.Equal(t, clusterID, actualClusterID)
+
+}
+// VerifyListProjects validates a user with the required cluster permissions are able/not able to list projects in the downstream cluster
+func VerifyListProjects(t *testing.T, client, standardClient *rancher.Client, clusterID, role, adminProjectName string) {
+	membersListProject, err := projects.ListProjectNames(standardClient, clusterID)
+	require.NoError(t, err)
+	projectlistAdmin, err := projects.ListProjectNames(client, clusterID)
+	require.NoError(t, err)
+
+	switch role {
+	case roleOwner, restrictedAdmin:
+		assert.Equal(t, len(projectlistAdmin), len(membersListProject))
+		assert.Equal(t, projectlistAdmin, membersListProject)
+	case roleMember:
+		assert.Equal(t, 0, len(membersListProject))
+	case roleProjectOwner, roleProjectMember:
+		assert.Equal(t, 1, len(membersListProject))
+		assert.Equal(t, adminProjectName, membersListProject[0])
+	}
+}
+
+// VerifyCreateProjects validates a user with the required cluster permissions are able/not able to create projects in the downstream cluster
+func VerifyCreateProjects(t *testing.T, client, standardClient *rancher.Client, clusterID, role string) {
+	memberProject, err := standardClient.Management.Project.Create(projects.NewProjectConfig(clusterID))
+	switch role {
+	case roleOwner, roleMember, restrictedAdmin:
+		require.NoError(t, err)
+		log.Info("Created project as a ", role, " is ", memberProject.Name)
+		require.NoError(t, err)
+		actualStatus := fmt.Sprintf("%v", memberProject.State)
+		assert.Equal(t, "active", actualStatus)
+	case roleProjectOwner, roleProjectMember:
+		require.Error(t, err)
+		errStatus := strings.Split(err.Error(), ".")[1]
+		rgx := regexp.MustCompile(`\[(.*?)\]`)
+		errorMsg := rgx.FindStringSubmatch(errStatus)
+		assert.Equal(t, "403 Forbidden", errorMsg[1])
+	}
+}
+
+// VerifyCreateNS validates a user with the required cluster permissions are able/not able to create namespaces in the project they do not own
+func VerifyCreateNS(t *testing.T, client, standardClient *rancher.Client, project *management.Project, clusterID, role string) {
+	var checkErr error
+	namespaceName := namegen.AppendRandomString("testns-")
+	standardClient, err := standardClient.ReLogin()
+	require.NoError(t, err)
+
+	createdNamespace, checkErr := namespaces.CreateNamespace(standardClient, namespaceName, "{}", map[string]string{}, map[string]string{}, project)
+
+	switch role {
+	case roleOwner, roleProjectOwner, roleProjectMember, restrictedAdmin:
+		require.NoError(t, checkErr)
+		log.Info("Created a namespace as role ", role, createdNamespace.Name)
+		assert.Equal(t, namespaceName, createdNamespace.Name)
+
+		namespaceStatus := &coreV1.NamespaceStatus{}
+		err = v1.ConvertToK8sType(createdNamespace.Status, namespaceStatus)
+		require.NoError(t, err)
+		actualStatus := fmt.Sprintf("%v", namespaceStatus.Phase)
+		assert.Equal(t, "Active", actualStatus)
```

Comment:
Tiny nit: Does "Active" need to be checked as case sensitive? Maybe make a const to cover the lowercase "active" on line 91 as well. And when checking use.

```go
strings.ToLower(actual)
```

Reply:
good point

Reply:
good point. Updated!